[CI] SingleValueQueryTests testNotMatchSome {p0=StandardSetup[fieldType=keyword, multivaluedField=true, empty=true, count=100]} failing

**Build scan:**
https://gradle-enterprise.elastic.co/s/ecs6aqcmqj57o/tests/:x-pack:plugin:esql:test/org.elasticsearch.xpack.esql.querydsl.query.SingleValueQueryTests/testNotMatchSome%20%7Bp0=StandardSetup%5BfieldType=keyword,%20multivaluedField=true,%20empty=true,%20count=100%5D%7D


**Reproduction line:**
```
./gradlew ":x-pack:plugin:esql:test" --tests "org.elasticsearch.xpack.esql.querydsl.query.SingleValueQueryTests.testNotMatchSome {p0=StandardSetup[fieldType=keyword, multivaluedField=true, empty=true, count=100]}" -Dtests.seed=82B29739D1818C51 -Dtests.locale=sr -Dtests.timezone=Pacific/Kosrae -Druntime.java=22
```

**Applicable branches:**
main

**Reproduces locally?:**
Didn't try

**Failure history:**
[Failure dashboard for `org.elasticsearch.xpack.esql.querydsl.query.SingleValueQueryTests#testNotMatchSome {p0=StandardSetup[fieldType=keyword, multivaluedField=true, empty=true, count=100]}`](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('testNotMatchSome%20%7Bp0%3DStandardSetup%5BfieldType%3Dkeyword,%20multivaluedField%3Dtrue,%20empty%3Dtrue,%20count%3D100%5D%7D'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.xpack.esql.querydsl.query.SingleValueQueryTests'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))


**Failure excerpt:**
```
java.lang.AssertionError: no warnings, expected: [WARN: Line -1:-1: evaluation of [] failed, treating result as null. Only first 20 failures recorded., WARN: Line -1:-1: java.lang.IllegalArgumentException: single-value function encountered multi-value]

  at __randomizedtesting.SeedInfo.seed([82B29739D1818C51:7BA7A52277433451]:0)
  at org.junit.Assert.fail(Assert.java:89)
  at org.junit.Assert.assertTrue(Assert.java:42)
  at org.junit.Assert.assertNotNull(Assert.java:713)
  at org.elasticsearch.test.ESTestCase.assertWarnings(ESTestCase.java:693)
  at org.elasticsearch.test.ESTestCase.assertWarnings(ESTestCase.java:662)
  at org.elasticsearch.xpack.esql.querydsl.query.SingleValueQueryTests.runCase(SingleValueQueryTests.java:156)
  at org.elasticsearch.xpack.esql.querydsl.query.SingleValueQueryTests.lambda$testNotMatchSome$4(SingleValueQueryTests.java:119)
  at org.elasticsearch.xpack.esql.querydsl.query.SingleValueQueryTests.testCase(SingleValueQueryTests.java:175)
  at org.elasticsearch.xpack.esql.querydsl.query.SingleValueQueryTests.testNotMatchSome(SingleValueQueryTests.java:117)
  at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
  at java.lang.reflect.Method.invoke(Method.java:580)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.junit.rules.RunRules.evaluate(RunRules.java:20)
  at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:48)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.junit.rules.RunRules.evaluate(RunRules.java:20)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
  at org.junit.rules.RunRules.evaluate(RunRules.java:20)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
  at java.lang.Thread.run(Thread.java:1570)

```

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -108,8 +108,6 @@ tests:
 - class: org.elasticsearch.datastreams.DataStreamsClientYamlTestSuiteIT
   method: test {p0=data_stream/190_failure_store_redirection/Ensure failure is redirected to correct failure store after a reroute processor}
   issue: https://github.com/elastic/elasticsearch/issues/111041
-- class: org.elasticsearch.xpack.esql.querydsl.query.SingleValueQueryTests
-  issue: https://github.com/elastic/elasticsearch/issues/111042
 - class: org.elasticsearch.compute.lucene.ValueSourceReaderTypeConversionTests
   method: testLoadAllStatusAllInOnePage
   issue: https://github.com/elastic/elasticsearch/issues/111048

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -97,9 +97,6 @@ tests:
 - class: org.elasticsearch.xpack.esql.qa.mixed.MixedClusterEsqlSpecIT
   method: "test {stats.Count_or_null SYNC #2}"
   issue: https://github.com/elastic/elasticsearch/issues/110950
-- class: "org.elasticsearch.xpack.esql.querydsl.query.SingleValueQueryTests"
-  issue: "https://github.com/elastic/elasticsearch/issues/110977"
-  method: "testNotMatchSome {p0=StandardSetup[fieldType=keyword, multivaluedField=true, empty=true, count=100]}"
 - class: org.elasticsearch.xpack.esql.qa.mixed.MixedClusterEsqlSpecIT
   issue: https://github.com/elastic/elasticsearch/issues/110978
 - class: org.elasticsearch.search.sort.FieldSortIT

--- a/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/querydsl/query/SingleValueMatchQuery.java
+++ b/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/querydsl/query/SingleValueMatchQuery.java
@@ -42,11 +42,10 @@ import java.util.Objects;
 final class SingleValueMatchQuery extends Query {
 
     /**
-     * The estimated number of comparisons to check if a {@link DocValues}
-     * has more than one value. There isn't a good way to get that number out of
-     * {@link DocValues} so this is a guess.
+     * We choose this value to big big so this approximation always run last. This avoid
+     * reporting warnings when queries are not matching multi-values
      */
-    private static final int MULTI_VALUE_MATCH_COST = 10;
+    private static final int MULTI_VALUE_MATCH_COST = 1000;
     private static final IllegalArgumentException MULTI_VALUE_EXCEPTION = new IllegalArgumentException(
         "single-value function encountered multi-value"
     );

--- a/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/querydsl/query/SingleValueMatchQuery.java
+++ b/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/querydsl/query/SingleValueMatchQuery.java
@@ -42,8 +42,8 @@ import java.util.Objects;
 final class SingleValueMatchQuery extends Query {
 
     /**
-     * We choose this value to big big so this approximation always run last. This avoid
-     * reporting warnings when queries are not matching multi-values
+     * Choose a big enough value so this approximation never drives the iteration.
+     * This avoids reporting warnings when queries are not matching multi-values
      */
     private static final int MULTI_VALUE_MATCH_COST = 1000;
     private static final IllegalArgumentException MULTI_VALUE_EXCEPTION = new IllegalArgumentException(

--- a/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/querydsl/query/SingleValueQueryTests.java
+++ b/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/querydsl/query/SingleValueQueryTests.java
@@ -148,8 +148,7 @@ public class SingleValueQueryTests extends MapperServiceTestCase {
         }
         assertThat(count, equalTo(expected));
 
-        // the SingleValueQuery.TwoPhaseIteratorForSortedNumericsAndTwoPhaseQueries can scan all docs - and generate warnings - even if
-        // inner query matches none, so warn if MVs have been encountered within given range, OR if a full scan is required
+        // we should only have warnings if we have matched a multi-value
         if (mvCountInRange > 0) {
             assertWarnings(
                 "Line -1:-1: evaluation of [] failed, treating result as null. Only first 20 failures recorded.",

--- a/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/querydsl/query/SingleValueQueryTests.java
+++ b/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/querydsl/query/SingleValueQueryTests.java
@@ -76,7 +76,7 @@ public class SingleValueQueryTests extends MapperServiceTestCase {
         int max = between(1, 100);
         testCase(
             new SingleValueQuery.Builder(new RangeQueryBuilder("i").lt(max), "foo", Source.EMPTY),
-            (fieldValues, count) -> runCase(fieldValues, count, null, max, false)
+            (fieldValues, count) -> runCase(fieldValues, count, null, max)
         );
     }
 
@@ -116,7 +116,7 @@ public class SingleValueQueryTests extends MapperServiceTestCase {
         int max = between(1, 100);
         testCase(
             new SingleValueQuery(new RangeQuery(Source.EMPTY, "i", null, false, max, false, null), "foo").negate(Source.EMPTY).asBuilder(),
-            (fieldValues, count) -> runCase(fieldValues, count, max, 100, true)
+            (fieldValues, count) -> runCase(fieldValues, count, max, 100)
         );
     }
 
@@ -132,10 +132,8 @@ public class SingleValueQueryTests extends MapperServiceTestCase {
      * @param count The count of the docs the query matched.
      * @param docsStart The start of the slice in fieldValues we want to consider. If `null`, the start will be 0.
      * @param docsStop The end of the slice in fieldValues we want to consider. If `null`, the end will be the fieldValues size.
-     * @param scanForMVs Should the check for Warnings scan the entire fieldValues? This will override the docsStart:docsStop interval,
-     *                   which is needed for some cases.
      */
-    private void runCase(List<List<Object>> fieldValues, int count, Integer docsStart, Integer docsStop, boolean scanForMVs) {
+    private void runCase(List<List<Object>> fieldValues, int count, Integer docsStart, Integer docsStop) {
         int expected = 0;
         int min = docsStart != null ? docsStart : 0;
         int max = docsStop != null ? docsStop : fieldValues.size();
@@ -152,7 +150,7 @@ public class SingleValueQueryTests extends MapperServiceTestCase {
 
         // the SingleValueQuery.TwoPhaseIteratorForSortedNumericsAndTwoPhaseQueries can scan all docs - and generate warnings - even if
         // inner query matches none, so warn if MVs have been encountered within given range, OR if a full scan is required
-        if (mvCountInRange > 0 || (scanForMVs && fieldValues.stream().anyMatch(x -> x.size() > 1))) {
+        if (mvCountInRange > 0) {
             assertWarnings(
                 "Line -1:-1: evaluation of [] failed, treating result as null. Only first 20 failures recorded.",
                 "Line -1:-1: java.lang.IllegalArgumentException: single-value function encountered multi-value"
@@ -161,7 +159,7 @@ public class SingleValueQueryTests extends MapperServiceTestCase {
     }
 
     private void runCase(List<List<Object>> fieldValues, int count) {
-        runCase(fieldValues, count, null, null, false);
+        runCase(fieldValues, count, null, null);
     }
 
     private void testCase(SingleValueQuery.Builder builder, TestCase testCase) throws IOException {

